spec/services/customers/manage_invoice_custom_sections_service_spec.rb

```diff
@@ -0,0 +1,123 @@
+    context 'when customer is not found' do
+      let(:customer) { nil }
+
+      it 'returns not found failure' do
+        result = service.call
+        expect(result).not_to be_success
+        expect(result.error).to be_a(BaseService::NotFoundFailure)
+        expect(result.error.message).to eq('customer_not_found')
+      end
+    end
+
+    context 'when sending skip_invoice_custom_sections: true AND selected_ids' do
+      let(:skip_invoice_custom_sections) { true }
+      let(:section_ids) { [1, 2, 3] }
+
+      it 'raises an error' do
+        result = service.call
+        expect(result).not_to be_success
+        expect(result.error).to be_a(BaseService::ValidationFailure)
+        expect(result.error.message).to include('skip_sections_and_selected_ids_sent_together')
+      end
+    end
+
+    context 'when updating selected_invoice_custom_sections' do
+      context 'when section_ids match customer\'s applicable sections' do
+        let(:section_ids) { [invoice_custom_sections.first.id] }
+
+        it 'returns the result without changes' do
+          result = service.call
+          expect(result).to be_success
+          expect(customer.applicable_invoice_custom_sections.ids).to match_array(section_ids)
+        end
+      end
+
+      context 'when section_ids match organization\'s selected sections' do
+        let(:section_ids) { invoice_custom_sections[2..3].map(&:id) }
+
+        it 'still sets selected invoice_custom_sections as custom' do
+          service.call
+          expect(customer.reload.selected_invoice_custom_sections.ids).to match_array(invoice_custom_sections[2..3].map(&:id))
+          expect(customer.applicable_invoice_custom_sections.ids).to match_array(section_ids)
+        end
+      end
+
+      context 'when section_ids are totally custom' do
+        let(:section_ids) { invoice_custom_sections[1..2].map(&:id) }
+
+        it 'assigns customer sections' do
+          service.call
+          expect(customer.reload.selected_invoice_custom_sections.ids).to match_array(section_ids)
+          expect(customer.applicable_invoice_custom_sections.ids).to match_array(section_ids)
+        end
+      end
+
+      context 'when selected_ids are an empty array' do
+        let(:section_ids) { [] }
+
+        it 'assigns organization sections' do
+          service.call
+          expect(customer.reload.selected_invoice_custom_sections.ids).to match_array([])
+          expect(customer.applicable_invoice_custom_sections.ids).to match_array(customer.organization.selected_invoice_custom_sections.ids)
+        end
+      end
+
+      context 'when setting invoice_custom_sections_ids when previously customer had skip_invoice_custom_sections' do
+        let(:section_ids) { [] }
+
+        before { customer.update(skip_invoice_custom_sections: true) }
+
+        it 'sets skip_invoice_custom_sections to false' do
+          service.call
+          expect(customer.reload.skip_invoice_custom_sections).to be_falsey
```

Comment:
here we want to ensure the value is `false` not falsey (nil evaluates as falsey for example)

```suggestion
          expect(customer.reload.skip_invoice_custom_sections).to eq false
```